### PR TITLE
Expose `ModuleInfo::prefix(&self)`

### PR DESCRIPTION
**Pull request by preston-evans98**
# Description
IBC needs to be able to identify the prefix where a module's data lives in the state trie. This PR exposes the necessary machinery.

### CODE REVIEW

File: module-system/sov-modules-api/src/lib.rs [hunk 1]
```diff
@@ -348,6 +348,9 @@ pub trait ModuleInfo {
     /// Returns address of the module.
     fn address(&self) -> &<Self::Context as Spec>::Address;
 
+    /// Returns address of the module.
```
Comment (bkolad) [inline module-system/sov-modules-api/src/lib.rs]: Should be "Returns prefix"

Comment (bkolad) [reply module-system/sov-modules-api/src/lib.rs:351]: Do we know why they need it?

**Comment by bkolad** [comment]
Do we know why IBC needs it?

**Review by bkolad — APPROVED** [review]

File: module-system/sov-modules-api/src/lib.rs (hunk 1 above)
Comment (preston-evans98) [reply module-system/sov-modules-api/src/lib.rs]: Fixed!

**Comment by preston-evans98** [comment]
> Do we know why IBC needs it?

Yeah, good question. The way IBC works is that messages from chain A to chain B get stored in a pre-defined region of the state trie of chain A. To receive messages, chain B checks a merkle proof that the message is at the expected path in A's state trie. But, IBC is meant to be pretty flexible - so chains can store the messages in whatever prefix of the state trie they want, and there's a "handshake" when you open a new connection where you tell the other chain your prefix.

**Pull request merged**